packages/flutter/lib/src/material/paginated_data_table.dart

```diff
@@ -432,32 +432,33 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
           child: Column(
             crossAxisAlignment: CrossAxisAlignment.stretch,
             children: <Widget>[
-              Semantics(
-                container: true,
-                child: DefaultTextStyle(
-                  // These typographic styles aren't quite the regular ones. We pick the closest ones from the regular
-                  // list and then tweak them appropriately.
-                  // See https://material.io/design/components/data-tables.html#tables-within-cards
-                  style: _selectedRowCount > 0 ? themeData.textTheme.subtitle1!.copyWith(color: themeData.accentColor)
-                                               : themeData.textTheme.headline6!.copyWith(fontWeight: FontWeight.w400),
-                  child: IconTheme.merge(
-                    data: const IconThemeData(
-                      opacity: 0.54
-                    ),
-                    child: Ink(
-                      height: 64.0,
-                      color: _selectedRowCount > 0 ? themeData.secondaryHeaderColor : null,
-                      child: Padding(
-                        padding: EdgeInsetsDirectional.only(start: startPadding, end: 14.0),
-                        child: Row(
-                          mainAxisAlignment: MainAxisAlignment.end,
-                          children: headerWidgets,
+              if (headerWidgets.isNotEmpty)
+                Semantics(
+                  container: true,
+                  child: DefaultTextStyle(
+                    // These typographic styles aren't quite the regular ones. We pick the closest ones from the regular
+                    // list and then tweak them appropriately.
+                    // See https://material.io/design/components/data-tables.html#tables-within-cards
+                    style: _selectedRowCount > 0 ? themeData.textTheme.subtitle1!.copyWith(color: themeData.accentColor)
```

Comment:
Can we replace accentColor with colorScheme.secondaryColor (a separate PR would be OK). The accentColor is no longer part of the Material Design spec - see "Deprecate/Remove the accent properties" in  flutter.dev/go/material-theme-system-updates

Reply:
Makes sense, I'll do it in the PR to bring it up to Material specs: https://github.com/flutter/flutter/issues/69766